rust/src/mlil/operation.rs

```diff
@@ -0,0 +1,1234 @@
+use binaryninjacore_sys::BNMediumLevelILInstruction;
+use binaryninjacore_sys::BNMediumLevelILOperation;
+
+use core::marker::PhantomData;
+
+use crate::{rc::Ref, types::SSAVariable};
+
+use super::*;
+
+pub struct Operation<O: OperationArguments> {
```

Comment:
If I'm reading this right:
 - Operations can return Expressions (Operation.left, operation.right, etc)
 - Expressions can return ExprLifted and ExprInfo
 - ExprLifed is the thing we like, it's what we can match on
 - ExprInfo....I can't tell what that's being used for

Can we remove `Expression`s and `ExperLifted` and just have operations return ExprInfo?
Also, if `Operation` objects are the "main thing you interact with," they should probably be renamed to `MediumLevelILInstruction`s

Reply:
> ExprLifed is the thing we like, it's what we can match on
> Can we remove Expressions and ExperLifted and just have operations return ExprInfo?

Do you want to remove or keep ExprLifed?

Can be removed, but you mentioned on slack:

> I think what it boils down to is being able to write match cases like `If(condition: LessThan(right: Add(lhs, rhs)))`

The main reason that I implemented both is:

`ExprLifted` is used for matching and is easy to use. Like in https://github.com/Vector35/binaryninja-api/tree/test_mlil_bindings/rust/src/mlil.

`ExprInfo` is useful if you need a more elegant code, is the equivalent of LLIL ExprInfo, only calling the API if the data is required, not preemptively allocate Vectors (like in ExprLifted) and is useful for implementing a visitor pattern.

Reply:
BTW ExprLifted is build using ExprInfo, It doesn't really make sense removing ExprInfo and keeping ExprLifted.

Reply:
Sorry, I misspoke there... I meant to say:
> Can we remove Expressions and ExprInfo and just have operations return ExprLifted?

In which case, can ExprLifted build itself? I don't actually see where ExperInfo is being used to make ExprLifted.